src/main/java/com/excelsiorjet/api/tasks/JetProject.java

```diff
@@ -643,13 +641,116 @@ public JetHome validate(boolean validateForBuild) throws JetTaskFailureException
+        }
+
+        for (DependencySettings dependencySettings : dependenciesSettings) {
+            if (dependencySettings.groupId == null && dependencySettings.artifactId == null) {
+                throw new JetTaskFailureException(s("JetApi.DependencySettingsWithoutGroupAndArtifact"));
+            }
+        }
+
+        for (DependencySettings externalDependency: externalDependencies) {
+            if (externalDependency.groupId != null || externalDependency.artifactId != null ||
+                    externalDependency.version != null) {
+                throw new JetTaskFailureException(s("JetApi.ExternalDependencyShouldNotHaveId", externalDependency.path));
+            }
+        }
+
+        for (DependencySettings dependencySettings: dependenciesSettings) {
+            if (dependenciesMatchedBy(dependencySettings).size() == 0) {
+                throw new JetTaskFailureException(s("JetApi.NoDependenciesForDependencySettings", dependencySettings.idStr()));
+            }
+        }
+
+        for (DependencySettings dep : dependencies) {
+            if (dep.packagePath != null && dep.disableCopyToPackage != null && dep.disableCopyToPackage) {
+                String msg = dep.isSettings() ?
+                        s("JetApi.DependencySettingsCannotHavePackagePathAndDisabledCopyToPackage", dep.idStr()) :
+                        s("JetApi.DependencyCannotHavePackagePathAndDisabledCopyToPackage", dep.path);
+                throw new JetTaskFailureException(msg);
+            }
+        }
+
+        Set<String> ids = new HashSet<>();
+        for (DependencySettings dependencySettings : dependenciesSettings) {
+            if (ids.contains(dependencySettings.idStr())) {
+                throw new JetTaskFailureException(s("JetApi.DuplicateDependencySettingsId", dependencySettings.idStr()));
+            }
+            ids.add(dependencySettings.idStr());
+        }
+
+        DependencySettingsResolver dependencySettingsResolver = new DependencySettingsResolver(groupId, dependenciesSettings);
+        if (appType() == ApplicationType.PLAIN) {
+            classpathEntries = new ArrayList<>();
+            if (mainJar != null) {
+                DependencySettings mainJarDep = new DependencySettings(mainJar, ProtectionType.ALL.userValue, OptimizationType.ALL.jetValue);
+                mainJarDep.packagePath = "";
+                classpathEntries.add(new ClasspathEntry(mainJarDep));
+            }
+            if (!ignoreProjectDependencies) {
+                for (ProjectDependency prjDep : projectDependencies) {
+                    classpathEntries.add(dependencySettingsResolver.resolve(prjDep));
+                }
+            }
+            for (DependencySettings extDep : externalDependencies) {
+                classpathEntries.add(new ClasspathEntry(extDep));
+            }
+        } else if (appType() == ApplicationType.TOMCAT && !ignoreProjectDependencies) {
+            classpathEntries = projectDependencies.stream().
```

Comment:
As I said earlier project artifactId should be taken to account here and we should also check that there is no external project dependencies for Tomcat